docs(contributing): FRMW-639-update bug and issue steps.

100stacks opened the pull request:
JIRA: https://jira.rax.io/browse/FRMW-639

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM

---
Update the process to submit bugs and issues to the `EncoreUI` team.

100stacks commented:
## Updated Steps to Report Bugs and Issues

<img width="1349" alt="screen shot 2016-01-28 at 3 19 47 pm" src="https://cloud.githubusercontent.com/assets/10120600/12659285/a720ddac-c5d3-11e5-96f5-c1912b574d00.png">

ssalinas24 commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,52 @@
+[GitHub Issues Tracker](https://github.com/rackerlabs/encore-ui/issues), with the following steps:
+
+* Open a new [GitHub Issue](https://github.com/rackerlabs/encore-ui/issues).
+* When submitting an issue:
```
@100stacks Do you think we could break the lists up and make this a lower header `### When submitting an issue:`? Then we could do the same for "When submitting a bug report:" below as well. @glynnis Would you agree?

100stacks commented:
@ssalinas24, updated based on your feedback. :+1:

glynnis commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,52 @@
+  - include applicable screenshots
+  - provide details about your browser and operating environment
+  - Framework version
+  - details about why/how it’s something that is blocking your team (if relevant), or how it could improve EncoreUI usage to 
```
Nitpicking here, but your first three points here start with a verb ("provide", "include", "provide"), but "Framework version" and "details about..." don't start with verbs. 

Maybe you can rephrase with the verb upfront, like
> When submitting an issue *please provide*:
> - the steps to reproduce...
> - applicable screenshots...

etc. so that the verb comes before the list.

glynnis replied:
To help with wordiness, you could change 
> details about why/how it’s something that is blocking your team (if relevant), or how it could improve EncoreUI usage > to help us prioritize team resources to address it

to 

> perspective on whether this issue is blocking your team (if relevant), or how it could improve EncoreUI usage (this 
> helps the Framework team prioritize your issue)

glynnis commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,52 @@
+  - rate the bug as minor or major
+* If applicable, the EncoreUI team will create a JIRA story to track the bug/issue that will be added to the current 
+or future sprint.
+* GitHub issue will remain open until addressed, and closed with steps taken by the EncoreUI team.
```
Change
> and closed with steps taken by the EncoreUI team

to
> and before closed, will show steps taken by the EncoreUI team.

ssalinas24 commented:
@100stacks Did you see the comments from Glynnis as well?

100stacks commented:
@glynnis @ssalinas24, feedback appreciated and incorporated. :+1: 

Updated screenshot provided in https://github.com/rackerlabs/encore-ui/pull/1519#issuecomment-176427592.

100stacks commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
 
 * Link to Invision designs (if applicable)
 * Screenshot of final visual design
-* For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design` label to request input
```
I updated the file to add line breaks at 121 characters or less.

glynnis commented:
LGTM

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+* Open a new [GitHub Issue](https://github.com/rackerlabs/encore-ui/issues).
+* When submitting a bug, please provide:
+  - **steps to reproduce the bug**.  If an EncoreUI developer cannot reproduce it, it 
+  becomes much more difficult to fix.
```
I'd update to use parens rather than a second sentence.

```
- **steps to reproduce the bug** (If an EncoreUI developer cannot reproduce it, 
  it becomes much more difficult to fix.)
```

CITguy replied:
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+  - details about your browser and operating environment
+  - Framework version
+  - perspective on whether this issue is blocking your team (if relevant), or how it could improve EncoreUI usage (this 
+helps the Framework team prioritize your issue)
```
Indent to same level as first character of first line.

```
  - perspective on whether this issue is blocking your team (if relevant), or how it could improve EncoreUI usage (this 
    helps the Framework team prioritize your issue)
```

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
-* For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design` label to request input
-* When all technical discussion on the issue is complete, change the label to `Ready for Dev`. For visual components/changes, *only* a designer may make this change.
+* For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design`
+label to request input
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+* For new visual components or visual changes, sign-off from our designers is _required_. Mark issue with `Needs Design`
+label to request input
+* When all technical discussion on the issue is complete, change the label to `Ready for Dev`. For visual 
+components/changes, *only* a designer may make this change.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -34,15 +69,18 @@ The following types of Pull Requests are most commonly seen:
     * A corresponding [Issue should be present](#preparing-an-issue-for-development)
-* **Step 1**: Submitter includes screenshot of new component in PR description (See ["Design Review of Pull Requests"](#design-review-of-pull-requests) below)
+* **Step 1**: Submitter includes screenshot of new component in PR description 
+(See ["Design Review of Pull Requests"](#design-review-of-pull-requests) below)
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -85,21 +123,28 @@ Create an issue for non-requested feedback & tag with types, for example:
 
-* Go to [http://khan.github.io/tota11y/](http://khan.github.io/tota11y/) and follow the installation option for dragging the tota11y button to your bookmarks bar.
+* Go to [http://khan.github.io/tota11y/](http://khan.github.io/tota11y/) and follow the installation option for dragging
+the tota11y button to your bookmarks bar.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -112,7 +157,8 @@ contrast is adequate, all checks on the page that Tota11y provides should show a
 * Include the "Before" and "After" images in your initial comment of the Pull Request.
-* Include a small description of the initial problem and solution displayed in the images. Including before and after shots show the contrast ratio improvements clearly to those reviewing your pull request.
+* Include a small description of the initial problem and solution displayed in the images. Including before and after 
+shots show the contrast ratio improvements clearly to those reviewing your pull request.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -136,19 +182,24 @@ on a pull request are as close to mock-ups as can be reasonably expected. Common
-    * Is color contrast compliant with A11y standards ([see "Request a check for color contrast with Tota11y" above](#using-tota11y-to-check-for-adequate-contrast))?
+    * Did the developer use colors provided by 
+    [the Framework's color palette](http://rackerlabs.github.io/encore-ui/#/styles/color), rather than sampling colors 
+    or introducing new hex codes?
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -136,19 +182,24 @@ on a pull request are as close to mock-ups as can be reasonably expected. Common
+    [the Framework's color palette](http://rackerlabs.github.io/encore-ui/#/styles/color), rather than sampling colors 
+    or introducing new hex codes?
+    * Is color contrast compliant with A11y standards 
+    ([see "Request a check for color contrast with Tota11y" above](#using-tota11y-to-check-for-adequate-contrast))?
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -136,19 +182,24 @@ on a pull request are as close to mock-ups as can be reasonably expected. Common
 * Consistency with similar elements or pages within the same application or within the Encore ecosystem
-    * Even when mock-ups are implemented and the screenshots match, there's still a possibility that the new work doesn't match existing precendents that have already been coded.
+    * Even when mock-ups are implemented and the screenshots match, there's still a possibility that the new work 
+    doesn't match existing precendents that have already been coded.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
-0. A pull request author finishes some new feature that contains visual changes *already mocked-up and approved by design*.
-0. The pull request author gathers the appropriate "before" (master branch, usually) and "after" screenshots of each relevant section.
+0. A pull request author finishes some new feature that contains visual changes *already mocked-up and approved by 
+design*.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
+0. A pull request author finishes some new feature that contains visual changes *already mocked-up and approved by 
+design*.
+0. The pull request author gathers the appropriate "before" (master branch, usually) and "after" screenshots of each 
+relevant section.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
-0. New screenshots are added, exactly how they were outlined above. Copy any pre-existing "before" screenshots from the old screenshot collection. Tag the designer reviewing this pull request.
-0. Previous screenshots are edited to no longer render inline. This prevents confusion while retaining accurate historical records of visual feedback during review.
+0. A comment is created [that tags an EncoreUI designer](https://github.com/blog/821). Immediately below this "@" 
+mention, the screenshots are [uploaded in the comment interface](https://help.github.com/articles/issue-attachments/).
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
+0. A comment is created [that tags an EncoreUI designer](https://github.com/blog/821). Immediately below this "@" 
+mention, the screenshots are [uploaded in the comment interface](https://help.github.com/articles/issue-attachments/).
+0. The initial visual feedback suggestion from our team recommends changing something so it better complies with 
+original mock-ups they provided. This change would require new screenshots!
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
+0. The initial visual feedback suggestion from our team recommends changing something so it better complies with 
+original mock-ups they provided. This change would require new screenshots!
+0. The pull request author changes something about the visual aspect of the design. They capture the new, changed 
+screenshots (the original ones remain unchanged).
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
+0. The pull request author changes something about the visual aspect of the design. They capture the new, changed 
+screenshots (the original ones remain unchanged).
+0. New screenshots are added, exactly how they were outlined above. Copy any pre-existing "before" screenshots from 
+the old screenshot collection. Tag the designer reviewing this pull request.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -214,24 +269,37 @@ Commit messages adhere to the Angular [commit message format](https://github.com
+0. New screenshots are added, exactly how they were outlined above. Copy any pre-existing "before" screenshots from 
+the old screenshot collection. Tag the designer reviewing this pull request.
+0. Previous screenshots are edited to no longer render inline. This prevents confusion while retaining accurate 
+historical records of visual feedback during review.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -247,18 +315,24 @@ Here is an example of how you should edit the old screenshot comments to change
-0. Update the latest master and do `git rebase master` on your branch, now that everything has been summed up into one or two commits
-    1. You may run into a merge conflict. In that case, open the conflicting file(s) and modify them so that they reflect the desired final state.
+0. `git rebase -i HEAD~x` where x is the number of commits you have made on the branch/PR (The "Commits" tab on the PR 
+page will display how many commits have been made)
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -247,18 +315,24 @@ Here is an example of how you should edit the old screenshot comments to change
+0. `git rebase -i HEAD~x` where x is the number of commits you have made on the branch/PR (The "Commits" tab on the PR 
+page will display how many commits have been made)
+0. Not including your original commit, mark `f`/`fixup` or `s`/`squash` for all commits after 
+it ([see example that follows](#example-of-step-2))
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -247,18 +315,24 @@ Here is an example of how you should edit the old screenshot comments to change
+0. Not including your original commit, mark `f`/`fixup` or `s`/`squash` for all commits after 
+it ([see example that follows](#example-of-step-2))
+0. Update the latest master and do `git rebase master` on your branch, now that everything has been summed up into one 
+or two commits
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -247,18 +315,24 @@ Here is an example of how you should edit the old screenshot comments to change
+0. Update the latest master and do `git rebase master` on your branch, now that everything has been summed up into one 
+or two commits
+    1. You may run into a merge conflict. In that case, open the conflicting file(s) and modify them so that they reflect 
+    the desired final state.
```
Indent to same level as first character of first line.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -85,21 +123,28 @@ Create an issue for non-requested feedback & tag with types, for example:
+contrast for critical text comply with
+standards, we use [Tota11y](http://khan.github.io/tota11y/), a bookmarklet tool that can be run in a development 
+environment. To demonstrate that a new feature or design complies, post an additional screen shot with Tota11y contrast 
+ratios showing along with the usual "before" and "after" screen shots. When
```
Please reflow this text. You can recombine text and create new line breaks

100stacks commented:
@CITguy, updated with your feedback. :+1:

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
 
-When submitting a bug report, please **include a set of steps to reproduce the issue** and any related information (browser, OS, etc).
-If an EncoreUI developer cannot reproduce the issue, it becomes much more difficult to fix.
+Please submit any bugs or issues you encounter when using EncoreUI to our 
```
Can we use "report" rather than "submit"? "submit" seems like we're *adding* an issue/bug, not *identifying* the issue/bug.

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+[GitHub Issues Tracker](https://github.com/rackerlabs/encore-ui/issues) following the steps below.
 
-### Preparing an Issue for development
+### How to Submit an Issue
```
`s/Submit/Report/`

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+### How to Submit an Issue
+
+* Open a new [GitHub Issue](https://github.com/rackerlabs/encore-ui/issues).
+* When submitting an issue, please provide:
```
`submitting` -> `reporting`

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+**Example Issue:** Please see a [reported issue](https://github.com/rackerlabs/encore-ui/issues/1465) that walks through 
+the steps of replicating, logging, tracking, and follow-up resolution with the reporter.  
+
+### How to Submit a Bug 
```
`Submit` -> `Report`

CITguy commented on CONTRIBUTING.md:
```diff
@@ -1,25 +1,60 @@
+### How to Submit a Bug 
+
+* Open a new [GitHub Issue](https://github.com/rackerlabs/encore-ui/issues).
+* When submitting a bug, please provide:
```
`submitting` -> `reporting`

100stacks commented:
@CITguy, updated screenshot in https://github.com/rackerlabs/encore-ui/pull/1519#issuecomment-176427592 with your comments. :+1: 

BTW, what happens if folks select this?
<img width="172" alt="screen shot 2016-02-01 at 5 03 43 pm" src="https://cloud.githubusercontent.com/assets/10120600/12734477/6214e3b2-c906-11e5-9f7e-c21ee6c67b1b.png">

CITguy commented:
@100stacks, click it and find out :)

CITguy commented:
LGTM

100stacks commented:
@CITguy, for https://github.com/rackerlabs/encore-ui/pull/1519#issuecomment-178247937 so it sends an email with browser and OS plus comments field.  Nice :wink: 

Do we get issues/bugs through here...or is it used more by the teams?

CITguy commented:
The "Submit Feedback" mechanism is more for consuming developers, though it doesn't stop people from using it to submit bugs.

parlarjb commented:
LGTM
